src/general/GameWorld.cs

```diff
@@ -310,6 +318,36 @@ public Species GetSpecies(uint id)
         return worldSpecies[id];
     }
 
+    /// <summary>
+    ///   Stores a description of a global event into the game world record.
+    /// </summary>
+    /// <param name="description">The event's description</param>
+    /// <param name="highlight">If true, the event will be highlighted in a timeline UI</param>
+    /// <param name="iconPath">Resource path to the icon of the event</param>
+    public void LogEvent(LocalizedString description, bool highlight = false, string iconPath = null)
+    {
+        if (eventsLog.Count > Constants.GLOBAL_EVENT_LOG_CAP)
+        {
+            var oldestKey = double.MaxValue;
+
+            foreach (var entry in eventsLog)
+                oldestKey = Math.Min(entry.Key, oldestKey);
```

Comment:
This looks a bit suspicious to me. I'd prefer either to do an entirely LINQ operation, I think eventsLog.Keys.Min() should work?, or code this "normally" with an if to check if the oldestKey variable needs to be updated or not.